packages/codegen-ui-react/lib/__tests__/__snapshots__/studio-ui-codegen-react-forms.test.ts.snap

```diff
@@ -4620,10 +4620,11 @@ export default function PostUpdateForm(props) {
             }
             promises.push(
               API.graphql({
-                query: deleteComment,
+                query: updateComment,
```

Comment:
Shouldn't this be conditional based on if the relationship is required or not? Generating a form that could never succeed with a delete doesn't really make sense. The error message would be very vague as well, especially for the downstream customers.

Reply:
We explicitly validate on if the relationship is required. if it is, then the update is prevented. This matches what is done today for forms using DataStore.

Reply:
Discussed over slack and concluded that this achieves parity with DataStore forms but we should consider follow up work to improve the customer experience when working with child collections.